src/main/java/seedu/address/ui/StatsWindow.java

```diff
@@ -0,0 +1,146 @@
+package seedu.address.ui;
+
+import javafx.fxml.FXML;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextAlignment;
+import javafx.scene.text.TextFlow;
+import javafx.stage.Stage;
+
+import javafx.stage.WindowEvent;
+import seedu.address.model.statistics.Statistics;
+
+/**
+ * Creates the statistics window that displays multiple statistics.
+ */
+public class StatsWindow extends UiPart<Stage> {
+    private static final String FXML = "StatsWindow.fxml";
+
+    private Statistics statistics;
+
+    @FXML
+    private HBox chartDisplayView;
+
+    @FXML
+    private HBox numDisplayView;
+
+    public StatsWindow(Stage root) {
+        super(FXML, root);
+
+        root.setOnCloseRequest((WindowEvent event) -> {
+            numDisplayView.getChildren().clear();
+            chartDisplayView.getChildren().clear();
+        });
+    }
+
+    public StatsWindow() {
+        this(new Stage());
+    }
+
+    /**
+     * Shows the statistics window.
+     */
+    public void show() {
+        getRoot().show();
+        getRoot().centerOnScreen();
+    }
+
+    /**
+     * Checks if the statistics window is currently showing.
+     */
+    public boolean isShowing() {
+        return getRoot().isShowing();
+    }
+
+    /**
+     * Hides the statistics window.
+     */
+    public void hide() {
+        getRoot().hide();
+    }
+
+    /**
+     * Focuses the statistics window if it's already open.
+     */
+    public void focus() {
+        getRoot().requestFocus();
+    }
+
+    /**
+     * Initialises the charts and graphs.
+     * @param statistics contains the data used to generate the charts and graphs.
+     */
+    public void initStats(Statistics statistics) {
```

Comment:
Running `stats` twice results in the elements being added twice, I think the `numDisplayView` and `chartDisplayView` should be cleared here to prevent that.

Reply:
oh!!! good catch